package NYT.NClickHouseServer.NProto;

import "yt/ytlib/chunk_client/proto/data_source.proto";
import "yt/ytlib/controller_agent/proto/job.proto";
import "yt_proto/yt/client/table_chunk_format/proto/chunk_meta.proto";

////////////////////////////////////////////////////////////////////////////////

message TSubquerySpec
{
    required NYT.NChunkClient.NProto.TDataSourceDirectoryExt data_source_directory = 1;
    repeated NYT.NControllerAgent.NProto.TTableInputSpec input_specs = 2;
    optional NYT.NTableClient.NProto.TTableSchemaExt read_schema = 6;
    optional int64 subquery_index = 8;
    optional int64 table_index = 9;
    optional string initial_query = 10;
    optional bytes table_reader_config = 11; // YSON-serialized TTableReaderConfig
    optional bytes query_settings = 12; // YSON-serialized TQuerySettings
};

message TSecondaryQueryReadTask
{
    repeated NYT.NControllerAgent.NProto.TTableInputSpec operand_inputs = 1;
};
